dashboards/resource_dashboard.go

```diff
@@ -175,9 +177,32 @@ func ResourceDashboard() common.Resource {
 			if err != nil {
 				return err
 			}
-			return w.Lakeview.Trash(ctx, dashboards.TrashDashboardRequest{
+
+			// Attempt to trash the dashboard.
+			err = w.Lakeview.Trash(ctx, dashboards.TrashDashboardRequest{
 				DashboardId: d.Id(),
 			})
+
+			// If the dashboard was already trashed, we'll get a 403 (Permission Denied) error.
+			// There may be other cases where we get a 403, so we first confirm that the
+			// dashboard state is actually trashed, and if so, return success.
+			if errors.Is(err, apierr.ErrPermissionDenied) {
```

Comment:
Thanks for the PR. Curious -- should we also check for 404 or if a dashboard is trashed already then the only response is 403? Mentioning since the api doc https://docs.databricks.com/api/workspace/lakeview/trash has 404 as one of the responses.

Reply:
It will return 404 if it really doesn't exist (not even in the trash).

Good point, though. This fix works around the issue at deletion time, but we should also treat trashed as a 404 at read time so that if it was trashed, the next plan will show **creation** instead of **recreation**.